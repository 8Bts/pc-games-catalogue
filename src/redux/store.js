import { createStore, applyMiddleware } from 'redux';
import promise from 'redux-promise-middleware';
import reducer from './reducers/index';

export default createStore(reducer, {}, applyMiddleware(promise));
