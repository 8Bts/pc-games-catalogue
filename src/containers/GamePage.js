import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { fetchSingleGameData } from '../redux/actions';
import singlegame from '../styles/singlegame.module.css';

const propTypes = {
  game: PropTypes.shape({
    name: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.object),
    description: PropTypes.string,
    released: PropTypes.string,
    img: PropTypes.string,
    website: PropTypes.string,
    rating: PropTypes.number,
    platforms: PropTypes.arrayOf(PropTypes.object),
  }),
  match: PropTypes.shape({
    params: PropTypes.shape({
      gameId: PropTypes.string.isRequired,
    }),
  }).isRequired,
  fetchSingleGameData: PropTypes.func.isRequired,
};

const defaultProps = {
  game: {
    name: '',
    genres: [],
    description: '',
    released: '',
    img: '',
    website: '',
    rating: -1,
    platforms: [],
  },
};

const GamePage = ({
  game,
  match,
  fetchSingleGameData,
}) => {
  useEffect(() => {
    fetchSingleGameData(match.params.gameId);
  }, []);

  const background = {
    backgroundImage: `url(${game.img})`,
  };
  const requirements = game.platforms.length > 0 ? game.platforms.find((obj) => obj.platform.name === 'PC').requirements : { minumum: null, recommended: null };
  return (
    <div>
      <div className={singlegame.page}>
        <header>
          <div className="header">
            <h1 className={`${singlegame.h1} text-center p-2`}>{game.name}</h1>
          </div>
        </header>
        <section>
          <div className="row gx-0 my-4">
            <div className={`${singlegame.img} col-md-5`} style={background} />
            <div className={`${singlegame.right} col-md-7 d-flex flex-column`}>
              <span className="mb-1 d-flex">
                <span>Genres:</span>
                <span className="d-flex flex-wrap align-items-center">
                  { game.genres.map((genre) => (<span className="badge bg-info m-2 mt-1" key={genre.id}>{genre.name}</span>))}
                </span>
              </span>
              <span className="mb-1">
                Rating:
                <span className="badge bg-warning mx-2">
                  {game.rating}
                  {' / 5'}
                </span>
              </span>
              <span className="my-3">
                Released:
                <i>
                  {` ${game.released}`}
                </i>
              </span>
              <address>
                Website:
                <a target="_blank" rel="noreferrer" href={game.website} className="link-success ps-2">{game.website}</a>
              </address>
              <span className={singlegame.verdana}>{requirements.minimum}</span>
              <span className={singlegame.verdana}>{requirements.recommended}</span>
            </div>
          </div>
          <div>
            <h2 className={`${singlegame.h2} text-center mb-3 p-1`}>Description</h2>
            <div className={singlegame.verdana}>{game.description}</div>
          </div>
        </section>
      </div>
    </div>
  );
};

GamePage.propTypes = propTypes;
GamePage.defaultProps = defaultProps;

const mapStateToProps = (state) => {
  if (!state.singleGame.name) return {};

  return {
    game: {
      name: state.singleGame.name,
      genres: state.singleGame.genres,
      description: state.singleGame.description_raw,
      released: state.singleGame.released,
      img: state.singleGame.background_image,
      website: state.singleGame.website,
      rating: state.singleGame.rating,
      platforms: state.singleGame.platforms,
    },
  };
};

const mapDispatchToProps = {
  fetchSingleGameData,
};

export default connect(mapStateToProps, mapDispatchToProps)(GamePage);
