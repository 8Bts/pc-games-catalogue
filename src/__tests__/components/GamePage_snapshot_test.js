import toJson from 'enzyme-to-json';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import GamePage from '../../containers/GamePage';
import store from '../../redux/store';

configure({ adapter: new Adapter() });
it('Renders correctly', () => {
  const match = {
    params: {
      gameId: '3498',
    },
  };

  const wrapper = mount(
    <Provider store={store}>
      <GamePage match={match} />
    </Provider>,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
