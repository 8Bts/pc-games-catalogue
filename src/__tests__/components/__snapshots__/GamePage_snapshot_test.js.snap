// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(GamePage)
        match={
          Object {
            "params": Object {
              "gameId": "3498",
            },
          }
        }
      >
        <GamePage
          fetchSingleGameData={[Function]}
          game={
            Object {
              "description": "",
              "genres": Array [],
              "img": "",
              "name": "",
              "platforms": Array [],
              "rating": -1,
              "released": "",
              "website": "",
            }
          }
          match={
            Object {
              "params": Object {
                "gameId": "3498",
              },
            }
          }
        >
          <div>
            <div
              className="page"
            >
              <header>
                <div
                  className="header"
                >
                  <h1
                    className="h1 text-center p-2"
                  >
                    <Link
                      className="back"
                      to="/"
                    >
                      <LinkAnchor
                        className="back"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="back"
                          href="/"
                          onClick={[Function]}
                        >
                          <span
                            className="d-none d-md-inline-block"
                          >
                            Back
                          </span>
                          <img
                            alt="Back"
                            className="backicon"
                            src="back.png"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </h1>
                </div>
              </header>
              <section>
                <div
                  className="row gx-0 my-4"
                >
                  <div
                    className="img col-md-5"
                    style={
                      Object {
                        "backgroundImage": "url()",
                      }
                    }
                  />
                  <div
                    className="right col-md-7 d-flex flex-column"
                  >
                    <span
                      className="mb-1 d-flex"
                    >
                      <span
                        className="fw-bold"
                      >
                        Genres:
                      </span>
                      <span
                        className="d-flex flex-wrap align-items-center"
                      />
                    </span>
                    <span
                      className="mb-1 fw-bold"
                    >
                      Rating:
                      <span
                        className="badge bg-warning mx-2"
                      >
                        -1
                         / 5
                      </span>
                    </span>
                    <span
                      className="my-3 fw-bold"
                    >
                      Released:
                      <i
                        className="ps-2 fw-normal"
                      >
                         
                      </i>
                    </span>
                    <address
                      className="fw-bold"
                    >
                      Website:
                      <a
                        className="link-success ps-2"
                        href=""
                        rel="noreferrer"
                        target="_blank"
                      />
                    </address>
                    <span
                      className="verdana"
                    />
                    <span
                      className="verdana"
                    />
                  </div>
                </div>
                <div>
                  <h2
                    className="h2 text-center mb-3 p-1"
                  >
                    Description
                  </h2>
                  <div
                    className="verdana"
                  />
                </div>
              </section>
            </div>
          </div>
        </GamePage>
      </Connect(GamePage)>
    </Router>
  </BrowserRouter>
</Provider>
`;
